packages/lib-classifier/src/store/FeedbackStore.js

```diff
@@ -1,45 +1,76 @@
 import { autorun } from 'mobx'
-import { addDisposer, getRoot, onAction, types } from 'mobx-state-tree'
+import { addDisposer, addMiddleware, getRoot, onAction, types } from 'mobx-state-tree'
+import { flatten } from 'lodash'
 
 import helpers from './feedback/helpers'
 import strategies from './feedback/strategies'
 
 const FeedbackStore = types
   .model('FeedbackStore', {
     isActive: types.optional(types.boolean, false),
-    rules: types.map(types.frozen({}))
+    rules: types.map(types.frozen({})),
+    showModal: types.optional(types.boolean, false),
+    allowSubjectAdvance: types.optional(types.boolean, false)
   })
+  .views(self => ({
+    get hideSubjectViewer () {
+      return flatten(Array.from(self.rules.values()))
+        .some(rule => rule.hideSubjectViewer)
+    },
+    get messages () {
+      return flatten(Array.from(self.rules.values()))
```

Comment:
I completely forgot that `rules` is a map. Is this flattening it down to one array of rules for the entire workflow?

Reply:
Yes. It's different than how it currently is in pfe, but since we're updating and showing all feedback on completion I think we want all appropriate messages. The list should probably label or provide something to note which task and question the feedback is related to though.

Reply:
Yes, and if it's a branching workflow like Galaxy Zoo, the array that's plucked out here will only contain messages for the tasks that were chosen?

Reply:
I think it would only return messages for tasks that were chosen based on the following conditional checks that include `rule.success` or `!rule.success`.